### PR TITLE
Removing device info and extending the user-agent with Cliqz version

### PR DESCRIPTION
**Pull request by spacifici**

### CODE REVIEW

**Comment by konark-cliqz** [comment]
Can you share the before and after UA string?

**Comment by spacifici** [comment]
Before:
`Mozilla/5.0 (Linux; Android 6.0; Android SDK built for x86_64 Build/MASTER; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/44.0.2403.119 Mobile Safari/537.36`

After:
`Mozilla/5.0 (Linux; Android 6.0; Generic Android Device; wv) Version/4.0 Chrome/44.0.2403.119 Mobile Safari/537.36 Cliqz/1.9.7-debug`

File: app/src/main/java/acr/browser/lightning/view/LightningView.java [hunk 1]
```diff
@@ -581,14 +587,38 @@ private String getMobileUserAgent() {
...
+            final String product = matcher.group(1);
+            final String systemAndPlatform = matcher.group(2);
+            final String extensions = matcher.group(3);
+            final String cleanSystemAndPlatform = getCleanSystemAndPlatform(systemAndPlatform);
+            userAgent = String.format(
+                    "%s (%s) %s Cliqz/%s",
+                    product,
+                    cleanSystemAndPlatform,
+                    extensions,
+                    BuildConfig.VERSION_NAME
```
Comment (chrmod) [inline app/src/main/java/acr/browser/lightning/view/LightningView.java:600]: Is there any good reason to include version in User Agent? Is version name just a product name or it include actual product version?

**Comment by chrmod** [comment]
> Before:
> `Mozilla/5.0 (Linux; Android 6.0; Android SDK built for x86_64 Build/MASTER; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/44.0.2403.119 Mobile Safari/537.36`
> 
> After:
> `Mozilla/5.0 (Linux; Android 6.0; Generic Android Device; wv) Version/4.0 Chrome/44.0.2403.119 Mobile Safari/537.36 Cliqz/1.9.7-debug`

Is the `Generic Android Device` is going to be there for all users or just for simulator? I would avoid sharing actual device name.

**Comment by konark-cliqz** [comment]
Adding more information in UA can make it possible to fingerprint the users, specially if we append more precise information like `Cliqz/1.9.7-debug`. We should keep it to as minimal as possible.

**Comment by spacifici** [comment]
> Is the `Generic Android Device` is going to be there for all users or just for simulator? I would avoid sharing actual device name.

Everybody, the default one has the specific model name

File: app/src/main/java/acr/browser/lightning/view/LightningView.java (hunk 1 above)
Comment (spacifici) [reply app/src/main/java/acr/browser/lightning/view/LightningView.java:600]: The format is <product>/<version>, to me version can be a generic 1.0.0

**Comment by spacifici** [comment]
> Adding more information in UA can make it possible to fingerprint the users, specially if we append more precise information like `Cliqz/1.9.7-debug`. We should keep it to as minimal as possible.



> Adding more information in UA can make it possible to fingerprint the users, specially if we append more precise information like `Cliqz/1.9.7-debug`. We should keep it to as minimal as possible.

We discussed this before. It is a way to make websites know us. As above, we can replace the version part with something more generic as `1.0.0`.

**Comment by chrmod** [comment]
> > Adding more information in UA can make it possible to fingerprint the users, specially if we append more precise information like `Cliqz/1.9.7-debug`. We should keep it to as minimal as possible.
> 
> We discussed this before. It is a way to make websites know us. As above, we can replace the version part with something more generic as `1.0.0`.

On iOS we add just ` Cliqz` postfix. (Note that even this single sting breaks some pages and then we have to remove it based on a fixed list)

**Comment by konark-cliqz** [comment]
> > > Adding more information in UA can make it possible to fingerprint the users, specially if we append more precise information like `Cliqz/1.9.7-debug`. We should keep it to as minimal as possible.
> > 
> > 
> > We discussed this before. It is a way to make websites know us. As above, we can replace the version part with something more generic as `1.0.0`.
> 
> On iOS we add just ` Cliqz` postfix. (Note that even this single sting breaks some pages and then we have to remove it based on a fixed list)

If I recall correctly, one of such cases was `www.miles-and-more.com`.

This throws 403:
curl -H ':authority:www.miles-and-more.com' -H 'user-agent:Mozilla/5.0 (iPhone; CPU iPhone OS 13_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.4  Mobile/15E148 Safari/605.1.15 Cliqz' -H 'accept:text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' -H 'accept-language:en-us' -H 'accept-encoding:gzip, deflate, br' -H 'content-length:0' 'https://www.miles-and-more.com/de/en/program/status-benefits/senator-status.html'

Works:

curl -H ':authority:www.miles-and-more.com' -H 'user-agent:Mozilla/5.0 (iPhone; CPU iPhone OS 13_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.4 Mobile/15E148 Safari/605.1.15 Cliqz' -H 'accept:text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' -H 'accept-language:en-us' -H 'accept-encoding:gzip, deflate, br' -H 'content-length:0' 'https://www.miles-and-more.com/de/en/program/status-benefits/senator-status.html'

**Comment by chrmod** [comment]
> > > > Adding more information in UA can make it possible to fingerprint the users, specially if we append more precise information like `Cliqz/1.9.7-debug`. We should keep it to as minimal as possible.
> > > 
> > > 
> > > We discussed this before. It is a way to make websites know us. As above, we can replace the version part with something more generic as `1.0.0`.
> > 
> > 
> > On iOS we add just ` Cliqz` postfix. (Note that even this single sting breaks some pages and then we have to remove it based on a fixed list)
> 
> If I recall correctly, one of such cases was `www.miles-and-more.com`.
> 
> This throws 403:
> curl -H ':authority:www.miles-and-more.com' -H 'user-agent:Mozilla/5.0 (iPhone; CPU iPhone OS 13_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.4 Mobile/15E148 Safari/605.1.15 Cliqz' -H 'accept:text/html,application/xhtml+xml,application/xml;q=0.9,_/_;q=0.8' -H 'accept-language:en-us' -H 'accept-encoding:gzip, deflate, br' -H 'content-length:0' 'https://www.miles-and-more.com/de/en/program/status-benefits/senator-status.html'
> 
> Works:
> 
> curl -H ':authority:www.miles-and-more.com' -H 'user-agent:Mozilla/5.0 (iPhone; CPU iPhone OS 13_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.4 Mobile/15E148 Safari/605.1.15 Cliqz' -H 'accept:text/html,application/xhtml+xml,application/xml;q=0.9,_/_;q=0.8' -H 'accept-language:en-us' -H 'accept-encoding:gzip, deflate, br' -H 'content-length:0' 'https://www.miles-and-more.com/de/en/program/status-benefits/senator-status.html'

I've noticed it on `lot.com` which is also airline stuff.

**Comment by spacifici** [comment]
Guys, can you please check the script I use to test? https://pastebin.com/gy37672r
Both miles-and-more and lot.com works with my changes.

**Pull request merged**